src/Dfe.PlanTech.Infrastructure.Contentful/Helpers/ContentfulSetup.cs

```diff
@@ -36,13 +36,20 @@ public static IServiceCollection SetupContentfulClient(this IServiceCollection s
 
         services.AddTransient<IGetSubmissionStatusesQuery, GetSubmissionStatusesQuery>();
         services.AddTransient<ILogger<Category>, Logger<Category>>();
-        services.AddScoped<IContentfulClient, ContentfulClient>();
+        services.AddScoped<IContentfulClient>((services) =>
+        {
+            var options = services.GetRequiredService<ContentfulOptions>();
+            var typedClientFactory = services.GetRequiredService<IHttpClientFactory>();
+            var httpClient = typedClientFactory.CreateClient("contentful");
+
+            return new ContentfulClient(httpClient, options);
+        });
         services.AddScoped<IContentRepository, ContentfulRepository>();
 
 
         services.SetupRichTextRenderer();
-
-        setupClient(services.AddHttpClient<ContentfulClient>());
+        services.AddScoped<CustomDelegatingHandlerTokenRefresher>();
+        setupClient(services.AddHttpClient<ContentfulClient>("contentful").AddHttpMessageHandler<CustomDelegatingHandlerTokenRefresher>());
```

Comment:
Remove line above + 
```suggestion
        setupClient(services.AddHttpClient<ContentfulClient>("contentful"));```

Reply:
[a7ee237](https://github.com/DFE-Digital/sts-plan-technology-for-your-school/pull/900/commits/a7ee23746b4af6efd015645300904b889a4e35c8)